mlflow/tracing/utils/__init__.py

```
@@ -221,11 +222,12 @@ def maybe_get_request_id(is_evaluate=False) -> Optional[str]:
         return None
 
     if not context.request_id and is_evaluate:
-        raise MlflowTracingException(
-            f"Missing request_id for context {context}. "
-            "request_id can't be None when is_evaluate=True.",
-            error_code=BAD_REQUEST,
+        _logger.warning(
+            f"Missing request_id for context {context}. request_id can't be None when "
+            "is_evaluate=True. This is likely an internal error of MLflow, please file "
+            "a bug report at https://github.com/mlflow/mlflow/issues."
```

I'm a bit worried that if this breaks users just ignore the warnings in the logs, any specific reason that we don't want to keep exception here?

---

This does not raise exception now, the trace export is wrapped with try-catch. We don't have any upstream that wants to raise exception from this function.

---

Ah I see, the try catch swallows the exception.